apps/snitch_api/lib/snitch_api_web/controllers/hosted_payment_controller.ex

```diff
@@ -0,0 +1,106 @@
+defmodule SnitchApiWeb.HostedPaymentController do
+  use SnitchApiWeb, :controller
+  alias SnitchApi.Payment.HostedPayment
+  alias SnitchPayments
+  alias SnitchPayments.Gateway.PayuBiz
+  alias SnitchPayments.Provider
+
+  plug(SnitchApiWeb.Plug.DataToAttributes)
+  action_fallback(SnitchApiWeb.FallbackController)
+
+  @base_url "http://localhost:3000/api/v1/hosted-payment/"
+  @frontend_url ""
+
+  def payubiz_request_url(conn, params) do
+    {params, url} = payubiz_params_setup(params)
+    params = Keyword.put(params, :hash, generate_payubiz_hash(params))
+
+    headers = [
+      {"Content-Type", "application/x-www-form-urlencoded"}
+    ]
+
+    {:ok, response} = HTTPoison.post(url, {:form, params}, headers, follow_redirect: false)
+
+    {_, location} = List.keyfind(response.headers, "Location", 0)
+    render(conn, "payubiz-url.json-api", url: location)
+  end
+
+  def payment_success(conn, params) do
+    response = SnitchPayments.data_parser(params)
+
+    with {:ok, _, _} <- HostedPayment.payment_order_context(response) do
+      redirect(conn, external: @frontend_url <> "?info=payment_success")
+    else
+      {:error, _} ->
+        redirect(conn, external: @frontend_url <> "?info=payment_sucess")
```

Comment:
It should  be error page I think

Reply:
This method is called when payment is successful. For both the cases mentioned here, the payment is successful from the gateway. The error case here refers to the point if there is some failure in updating the records. I think we decided to show the user success only and handle this on our side.